src/lib/Libdb/db_postgres_impl.c

```diff
@@ -867,41 +833,71 @@ pbs_db_save_obj(pbs_db_conn_t *conn, pbs_db_obj_info_t *obj, int savetype)
  *
  */
 int
-pbs_db_delete_attr_obj(pbs_db_conn_t *conn, pbs_db_obj_info_t *obj, void *obj_id, pbs_db_attr_list_t *attr_list) {
-	return (db_fn_arr[obj->pbs_db_obj_type].pg_db_del_attr_obj(conn, obj, obj_id, attr_list));
+pbs_db_delete_attr_obj(pbs_db_conn_t *conn, pbs_db_obj_info_t *obj, void *obj_id, char *sv_time, pbs_db_attr_list_t *cache_attr_list, pbs_db_attr_list_t *db_attr_list) 
+{
+
+	/* are there attributes to remove from memory / local cache? */
+	if (cache_attr_list->attr_count > 0) {
+		if (dist_cache_del_attrs(obj_id, cache_attr_list) != 0)
+			return -1;
+	}
+	return (db_fn_arr[obj->pbs_db_obj_type].pg_db_del_attr_obj(conn, obj_id, sv_time, db_attr_list));
 }
 
 /**
  * @brief
- *	Add/update attributes of an object to the database
+ *	recover all attributes from the distribute cache for the object id provided
  *
- * @param[in]	conn - Connected database handle
- * @param[in]	pbs_db_obj_info_t - Wrapper object that describes the object
- * @param[in]	id - Object id
- * @param[in]	attr_list - list of attributes to add/update
+ * @param[in]	id  - string object id
+ * @param[in]	last_savetm - load attributes that changed from the provided time 
+ * @param[out]   attr_list  - return the list of retrieved attributes
  *
  * @return      Error code
- * @retval       0  - success
- * @retval      -1  - Failure
- *
+ * @retval	 0 - success
+ * @retval	-1 - failure
  */
-int
-pbs_db_add_update_attr_obj(pbs_db_conn_t *conn, pbs_db_obj_info_t *obj, void *obj_id, pbs_db_attr_list_t *attr_list) {
-	return (db_fn_arr[obj->pbs_db_obj_type].pg_db_add_update_attr_obj(conn, obj, obj_id, attr_list));
+int 
+dist_cache_recov_attrs(char *id, char *last_savetm, pbs_db_attr_list_t *attr_list)
+{
+	attr_list->attr_count = 0;
+	CLEAR_HEAD(attr_list->attrs);
+	return 0;
 }
 
+
 /**
  * @brief
- *	Frees allocate memory of an Object
- *
- * @param[in]	obj - db object
+ *	save specified attributes to the distribute cache for the object id provided
  *
- * @return None
+ * @param[in]	id  - string object id
+ * @param[in]   attr_list  -  list of retrieved attributes to be saved
  *
+ * @return      Error code
+ * @retval	 0 - success
+ * @retval	-1 - failure
  */
-void
-pbs_db_reset_obj(pbs_db_obj_info_t *obj)
+int 
+dist_cache_save_attrs(char *id, pbs_db_attr_list_t *attr_list)
 {
-	db_fn_arr[obj->pbs_db_obj_type].pg_db_reset_obj(obj);
+	return 0;
+}
 
+
+/**
+ * @brief
+ *	delete specified attributes to the distribute cache for the object id provided
+ *
+ * @param[in]	id  - string object id
+ * @param[in]   attr_list  -  list of retrieved attributes to be deleted
+ *
+ * @return      Error code
+ * @retval	 0 - success
+ * @retval	-1 - failure
+ */
+int
+dist_cache_del_attrs(char *id, pbs_db_attr_list_t *attr_list)
+{
+	attr_list->attr_count = 0;
+	CLEAR_HEAD(attr_list->attrs);
```

Comment:
Is it enough to just clear the head pointer. Just wanted to know don't we need to free all the attributes as well.

Reply:
It is not enough. It will be implemented when the dist cache is actually implemented. This is just a placeholder for now.